### PR TITLE
build: update ci job to use cargo-cross

### PR DESCRIPTION
**Pull request by fujiapple852**
WIP PR to support cargo cross (for netbsd).  See https://github.com/Byron/trash-rs/pull/87

cc @0323pin @Byron

### CODE REVIEW

**Comment by fujiapple852** [comment]
Fails to link for `netbsd` with:

```
/usr/local/lib/gcc/x86_64-unknown-netbsd/9.4.0/../../../../x86_64-unknown-netbsd/bin/ld: cannot find -lexecinfo
```

I'm not familiar with `libexecinfo` but seems to be needed here.

**Comment by fujiapple852** [comment]
Seemed to be this: https://github.com/cross-rs/cross/issues/1345

Pushed a fix

**Comment by fujiapple852** [comment]
This works when I change from `cargo test` to `cargo build` so I think there is a genuine issue with the tests on `netbsd` (I see there are several platform specific tests, I haven't dug into the logic much).  

I'd suggest trying to reproduce the test failure on a `netbsd` box and fixing up whatever is wrong.  Failing that we can update the pr to have a `cargo test` for all platforms and a `cargo build` for netbsd.

**Review by Byron — CHANGES_REQUESTED** [review]
Thanks a lot for getting NetBSD testing started!

Maybe for now, like you suggested, you could parameterize the `test` portion in `cargo test` so it can be `build` with NetBSD and `test` with everything else.

That way we wouldn't loose any test-coverage, validate builds in NetBSD while allowing tests to be enabled by contribution (CC @0323pin maybe) later.

**Comment by fujiapple852** [comment]
I ran the tests locally in my `netbsd` vm (9.3) and they pass.  So the issue here seem to be some combination of `cargo cross` and and `netbsd` and `cargo test`.

File: Cargo.toml [hunk 1]
```diff
@@ -51,3 +51,14 @@ windows = { version = "0.44.0", features = [
     "Win32_UI_Shell_PropertiesSystem",
 ] }
 scopeguard = "1.2.0"
+
+# workaround for https://github.com/cross-rs/cross/issues/1345
```
Comment (Byron) [inline Cargo.toml:55]: I hope we can keep an eye on this issue to eventually remove the workaround.

**Review by Byron — APPROVED** [review]
Thanks a lot for your help, it's much appreciated!

**Comment by 0323pin** [comment]
🚀 ❤️

**Pull request merged**